sdk/core/azure-core/test/ut/url.cpp

```diff
@@ -0,0 +1,255 @@
+  }
+
+  TEST(URL, query_parameter_encode_decode)
+  {
+    Http::HttpMethod httpMethod = Http::HttpMethod::Put;
+    Http::Url url("http://test.com");
+    EXPECT_NO_THROW(url.AppendQueryParameter("query", Http::Url::Encode("va=lue")));
+
+    // Default encoder from URL won't encode an equal symbol
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        url.GetAbsoluteUrl(),
+        "http://test.com?query=va%3Dlue");
+
+    // Provide symbol to do not encode
+    EXPECT_NO_THROW(url.AppendQueryParameter("query", Http::Url::Encode("va=lue", "=")));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        url.GetAbsoluteUrl(),
+        "http://test.com?query=va=lue");
+
+    // Provide more than one extra symbol to be encoded
+    EXPECT_NO_THROW(url.AppendQueryParameter("query", Http::Url::Encode("va=l u?e", " ?")));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        url.GetAbsoluteUrl(),
+        "http://test.com?query=va%3Dl u?e");
+
+    // default, encode it all
+    EXPECT_NO_THROW(url.AppendQueryParameter("query", Http::Url::Encode("va=l u?e")));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        url.GetAbsoluteUrl(),
+        "http://test.com?query=va%3Dl%20u%3Fe");
+  }
+
+  TEST(URL, add_path)
+  {
+    Http::HttpMethod httpMethod = Http::HttpMethod::Post;
+    Http::Url url("http://test.com");
+    Http::Request req(httpMethod, url);
+
+    EXPECT_NO_THROW(req.GetUrl().AppendPath("path"));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        req.GetUrl().GetAbsoluteUrl(),
+        "http://test.com/path");
+
+    EXPECT_NO_THROW(req.GetUrl().AppendQueryParameter("query", "value"));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        req.GetUrl().GetAbsoluteUrl(),
+        "http://test.com/path?query=value");
+
+    EXPECT_NO_THROW(req.GetUrl().AppendPath("path2"));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        req.GetUrl().GetAbsoluteUrl(),
+        "http://test.com/path/path2?query=value");
+
+    EXPECT_NO_THROW(req.GetUrl().AppendPath("path3"));
+    EXPECT_PRED2(
+        [](std::string a, std::string b) { return a == b; },
+        req.GetUrl().GetAbsoluteUrl(),
+        "http://test.com/path/path2/path3?query=value");
+  }
+
+  TEST(URL, getPort)
+  {
+    Http::Url url("http://test.com:9090");
+    uint16_t expected = 9090;
+
+    EXPECT_PRED2(
+        [](int expectedValue, int value) { return expectedValue == value; },
+        url.GetPort(),
+        expected);
+  }
+
+  TEST(URL, getPortConst)
+  {
+    Http::Url const url("https://test.com:500");
+    uint16_t expected = 500;
+
+    EXPECT_PRED2(
+        [](uint16_t expectedValue, uint16_t code) { return expectedValue == code; },
+        url.GetPort(),
+        expected);
+  }
```

Comment:
This test seems redundant (its the same as `getPort`), remove?

Reply:
noup, this is for a const url.  We are testing that GetPort is available for both const and non const url

Reply:
Good point. Missed that.